doc/api.md

```diff
@@ -0,0 +1,70 @@
+# `pub.dev` API for developers
+
+## Stay informed
+
+This document describes the officially supported API of the `pub.dev` site.
+`pub.dev` may expose API endpoints that are available publicly, but unless
+they are documented here, we don't consider them as officially
+supported, and may change or remove them without any notice.
+
+Changes to the officially supported `pub.dev` API will be announced at [Dart announce][announce].
+
+## Hosted Pub Repository API
+
+`pub.dev` implements the [Hosted Pub Repository Specification V2][repo-v2],
+used by the `pub` command line client application.
+
+## Additional APIs
+
+### Package names for name completion
+
+**GET** `https://pub.dev/api/package-name-completion-data`
+
+**Headers:**
+* `accept-encoding: gzip`
+
+**Response**
+* `cache-control: public, max-age=28800`
+* `content-encoding: gzip`
+* `content-type: application/json; charset="utf-8"`
+
+```js
+{
+  "packages": [
+    "http",
+    "provider",
+    /* further package names */
+  ]
+}
+```
+
+The API returns the top package names on `pub.dev`. At the moment we
+limit the result length at the maximum of 20000 package names, this
+limit may change in the future.
+
+The response is always a gzip-ed JSON content, and should be cached
+on the client side  for at least 8 hours between requests (as indicated
+by the `cache-control` header).
+
+Notes:
+ * Not all package names are included in this response.
+ * The inclusion criteria used by `pub.dev` may change without any early notice.
```

Comment:
```suggestion
The API returns the top package names on `pub.dev`. To reduce payload size the
result may not include all package names. The size limitation is subject to change.

The response is always a gzip-ed JSON content, and should be cached
on the client side  for at least 8 hours between requests (as indicated
by the `cache-control` header).

Notes:
 * Not all package names are included in this response.
 * The inclusion criteria used by `pub.dev` may change without notice.
```

Reply:
Or something like htat... I would suggest avoiding terms like "We"...  And avoid being too specific in describing what the API does.